UserService/UserService.BLL/Services/AuthService.cs

```diff
@@ -0,0 +1,253 @@
+
+            await cacheService.SetAsync(
+                CacheKeysProvider.GetRefreshKey(user.Id),
+                newRefreshToken,
+                TimeSpan.FromDays(30),
+                cancellationToken);
+
+            return new TokenResponse(newAccessToken, newRefreshToken);
+        }
+
+        public async Task<Guid> GenerateNewActivateCodeAsync(Guid id, CancellationToken cancellationToken = default)
+        {
+            var userEntity = await userRepository.GetByIdAsync(id, cancellationToken); 
+
+            if (userEntity is null)
+            {
+                throw new NotFoundException("User with the specified ID does not exist.");
+            }
+
+            if (userEntity.IsActivated)
+            {
+                throw new BadRequestException("The account is already activated.");
+            }
+
+            var activateCode = CodeProvider.GenerateSixDigitCode();
+
+            await cacheService.SetAsync(
+                CacheKeysProvider.GetActivateKey(userEntity.Id),
+                activateCode,
+                TimeSpan.FromMinutes(1),
+                cancellationToken);
+
+            //I'll move it to an event later
+            await emailService.SendEmail(new EmailDto
+            {
+                To = userEntity.Email,
+                Subject = "Activation code",
+                Body = activateCode.ToString()
+            },
+            cancellationToken);
+
+            return userEntity.Id;
+        }
+
+        public async Task<Guid> RegisterAsync(User newUser, CancellationToken cancellationToken = default)
+        {
+            var user = await userRepository
+                .GetByEmailAsync(newUser.Email, cancellationToken);
+
+            if(user is not null)
+            {
+                throw new BadRequestException("User with such email already exist");
+            }
+
+            var role = await roleRepository.GetByNameAsync(RoleEnum.User.ToString(), cancellationToken);
+
+            if (role is null)
+            {
+                throw new Exception();
+            }
+
+            newUser.Id = Guid.NewGuid();
+            newUser.RoleId = role.Id;
+
+            newUser.PasswordHash = passwordHasher.Generate(newUser.PasswordHash);
+
+            var userEntity = mapper.Map<UserEntity>(newUser);
+
+            await userRepository.CreateAsync(userEntity, cancellationToken);
+
+            var activateCode = CodeProvider.GenerateSixDigitCode();
+
+            await cacheService.SetAsync(
+                CacheKeysProvider.GetActivateKey(userEntity.Id),
+                activateCode,
+                TimeSpan.FromMinutes(1),
+                cancellationToken);
+
+            //I'll move it to an event later
+            await emailService.SendEmail(new EmailDto
+            {
+                To = userEntity.Email,
+                Subject = "Activation code",
+                Body = activateCode.ToString()
+            },
+            cancellationToken);
```

Comment:
I think this is code from the previous method, so you can move it to a separate method